samples/bluetooth/tmap_peripheral/src/bap_unicast_sr.c

```diff
@@ -212,46 +213,63 @@ static bool valid_metadata_type(uint8_t type, uint8_t len)
 	}
 }
 
-static int lc3_metadata(struct bt_bap_stream *stream, const struct bt_audio_codec_data *meta,
-			size_t meta_count, struct bt_bap_ascs_rsp *rsp)
-{
-	printk("Metadata: stream %p meta_count %u\n", stream, meta_count);
-	bool stream_context_present = false;
+struct data_func_param {
+	struct bt_bap_ascs_rsp *rsp;
+	bool stream_context_present;
+};
 
-	for (size_t i = 0; i < meta_count; i++) {
-		const struct bt_audio_codec_data *data = &meta[i];
+static bool data_func_cb(struct bt_data *data, void *user_data)
+{
+	struct data_func_param *func_param = (struct data_func_param *)user_data;
 
-		if (!valid_metadata_type(data->data.type, data->data.data_len)) {
-			printk("Invalid metadata type %u or length %u\n",
-			       data->data.type, data->data.data_len);
-			*rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_METADATA_REJECTED,
-					       data->data.type);
+	if (!valid_metadata_type(data->type, data->data_len)) {
+		printk("Invalid metadata type %u or length %u\n", data->type, data->data_len);
```

Comment:
```suggestion
		printk("Invalid metadata type %u or length %zu\n", data->type, data->data_len);
```

Reply:
There are more similar in the rest of the PR - no need to flood the comment thread ;)